DiscordBot.DataAccess/EventsSheetsService.cs

```diff
@@ -360,6 +355,92 @@ private int GetMetadataSheetId()
+        {
+            return new CellData()
+            {
+                UserEnteredValue = new ExtendedValue() { NumberValue = largestKey }
+            };
+        }
+
+        private Request MakeAddMetadataRequest(int eventKey, string name, string description, DateTime time)
+        {
+            var appendCellsRequest = new AppendCellsRequest()
+            {
+                Rows = new[]
+                {
+                    new RowData()
+                    {
+                        Values = new List<CellData>()
+                        {
+                            MakeCellData(eventKey),
+                            MakeCellData(name),
+                            MakeCellData(description),
+                            MakeCellData(time.ToString("s")),
+                            MakeCellData("Europe/London")
+                        }
+                    }
+                },
+                SheetId = metadataSheetId,
+                Fields = "*"
+            };
+
+            return new Request() { AppendCells = appendCellsRequest };
+        }
+
+        private Request MakeAddResponseSheetRequest(int eventKey)
+        {
+            var addResponseSheet = new AddSheetRequest()
+            {
+                Properties = new SheetProperties()
+                {
+                    Title = eventKey.ToString(),
+                    SheetId = eventKey
+                }
+            };
+
+            return new Request() { AddSheet = addResponseSheet };
+        }
+
+        private Request MakeAddResponseColumnsRequest(List<EventResponse> responseList)
+        {
+            var addResponseColumns = new AppendCellsRequest()
+            {
+                Rows = new[]
+                {
+                    new RowData()
+                    {
+                        Values = responseList
+                            .Select(response => response.Emoji)
+                            .Prepend("User id / Response emoji")
+                            .Select(MakeCellData)
+                            .ToList()
+                    },
+                    new RowData()
+                    {
+                        Values = responseList
+                            .Select(response => response.ResponseName)
+                            .Prepend("Response name:")
+                            .Select(MakeCellData)
+                            .ToList()
+                    }
+                },
+                SheetId = largestKey,
+                Fields = "*"
+            };
+
+            return new Request() { AppendCells = addResponseColumns };
+        }
+
+        private CellData MakeCellData(string stringValue)
+        {
+            return new CellData()
+            {
+                UserEnteredValue = new ExtendedValue() { StringValue = stringValue }
+            };
+        }
+
+
         private async Task<int> GetEventRowNumber(int eventKey)
         {
             try
```

Comment:
This file is quite long now. Might be worth trying to split it up, e.g. by extracting the methods to create request objects to a helper class? 🙂

Reply:
Yeah I agree. I've extracted those request making methods in db6b403 🙂